hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestStructuredStreaming.scala

```diff
@@ -207,12 +209,26 @@ class TestStructuredStreaming extends HoodieClientTestBase {
       metaClient.reloadActiveTimeline()
       assertEquals(1, getLatestFileGroupsFileId(HoodieTestDataGenerator.DEFAULT_FIRST_PARTITION_PATH).size)
     }
-    structuredStreamingForTestClusteringRunner(sourcePath, destPath, true,
+    structuredStreamingForTestClusteringRunner(sourcePath, destPath, true, false,
       HoodieTestDataGenerator.DEFAULT_FIRST_PARTITION_PATH, checkClusteringResult)
   }
 
   @Test
-  def testStructuredStreamingWithoutInlineClustering(): Unit = {
+  def testStructuredStreamingWithAsyncClustering(): Unit = {
```

Comment:
not sure if we this will be too hard to achieve. Is there a way to simulate resource unavailability. i.e. when async clustering is scheduled, no resources to schedule right away. But after you open up resources in your test, async clustering should get triggered. basically to validate that a pending async clustering should get triggered when resources become available and not get cancelled.

Reply:
I will try to simulate this as part of integration test. https://issues.apache.org/jira/browse/HUDI-1077